### ISSUE TITLE
Controller not recognized by games

### ISSUE THREAD

**Issue by frankruegamer** [riking/joycon#39]
I can connect both Joy-Cons just fine, so that both of them have one steady light. But after this I don't know how to proceed, it wasn't working in any game I've tried so far, though some games (i.e. Enter the Gungeon) showed that there was a controller connected.

Am I missing something?

**Comment by nickell-andrew** [comment]
Can you give more information about your setup? Are you using this with OSX, Windows, Linux? Which flavor/version?

Random Disclaimer: joycon's functionality in OSX is little more than a proof of concept as no one has written a driver to expose the joined joycons as a single controller to other software on an OSX system.

**Comment by frankruegamer** [comment]
I'm using Arch Linux. To add to that, testing tools like jstest appear to be able to detect buttons pressed and sticks moved (at least with the left joy con).

**Comment by nickell-andrew** [comment]
I don't know how helpful my input will be then @frankruegamer . I've only played around on OSX.

While I can't speak to integration inside games, I can say that the driver when running in a terminal successfully created a composite controller after I had both controllers connected via bluetooth.

Are you sure you have both joy con's successfully connected individually?

**Comment by frankruegamer** [comment]
@nickell-andrew I'm not quite sure what you mean by "connected individually". I have connected them both at the same time and one after the other so that eventually they stop blinking. Here is the output:
```
[console]> [INFO] Connected to Joy-Con L dc:68:eb:2a:87:b1
Joy-Con L (dc:68:eb:2a:87:b1): 🔋 █ 
dc:68:eb:2a:87:b1: Got factory calibration and button colors
dc:68:eb:2a:87:b1: SPI read returned [603d+25]
00000000  14 05 47 98 77 8f 4f a5  44 ff ff ff ff ff ff ff  |..G.w.O.D.......|
00000010  ff ff ff 0a b9 e6 00 1e  1e                       |.........|
dc:68:eb:2a:87:b1: SPI read returned [8010+22]
00000000  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
00000010  ff ff ff ff ff ff                                 |......|
dc:68:eb:2a:87:b1: Checked user stick calibration: [{1300 1136 1944 2295 1359 1098} {4095 4095 4095 4095 4095 4095}]
[INFO] Connected to Joy-Con R dc:68:eb:2a:6a:6b
Joy-Con R (dc:68:eb:2a:6a:6b): 🔋 █ 
dc:68:eb:2a:6a:6b: Got factory calibration and button colors
dc:68:eb:2a:6a:6b: SPI read returned [603d+25]
00000000  ff ff ff ff ff ff ff ff  ff 4d e8 70 f3 54 42 03  |.........M.p.TB.|
00000010  55 49 ff ff 3c 28 1e 0a  0a                       |UI..<(...|
dc:68:eb:2a:6a:6b: SPI read returned [8010+22]
00000000  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
00000010  ff ff ff ff ff ff                                 |......|
dc:68:eb:2a:6a:6b: Checked user stick calibration: [{4095 4095 4095 4095 4095 4095} {1283 1173 2125 1806 1267 1061}]
Plonk! (u1)
Plonk! (u0)
Plonk! (u1)
Plonk! (u0)
Plonk! (u1)
Plonk! (u1)
Plonk! (u0)
Plonk! (u0)
Plonk! (u1)
Plonk! (u0)
pairing double
set permissions for event25
set permissions for js2
```
When issuing the `list` command it shows that both joycons are connected
```
Connected JoyCons:
  c1l: dc:68:eb:2a:87:b1 🔋 █ 
  c1r: dc:68:eb:2a:6a:6b 🔋 █ 
```

**Comment by riking** [comment]
What does html5gamepad.com show? Sometimes, it's just that the game has weird detection rules and you need to tweak the output config per game (another reason to have actual config files).

**Comment by frankruegamer** [comment]
This site is able to detect every input (except gyro and screenshot) from my joycons. I tried it in multiple games (including Dead Cells, Enter the Gungeon, Tomb Rider, Wizard of Legend, Hyper Light Drifter, ...); while most of them were able to detect a controller, none would accept input from them. Two of them, Tomb Rider and Hyper Light Drifter, detected a fast and constant movement of a stick so it would switch menu entries really quickly. They work just fine with my PS3 controller.

**Comment by riking** [comment]
Yeah, so that means they don't like the output profile of the controller. Unfortunately, I don't really have time right now to work on creating a configuration file for that :(

**Comment by JPHanke** [comment]
Hi all, I'm facing the same issue... I can configure the joy cons as one single joystick in my retropie, but the games don't recognize it. Is there a solution for it? Or can you give us some directions so we can create our own config file? Thanks.

**Comment by Davidc999** [comment]
Your issue with retropie is probably that it detects 3 controllers: L, R and the full joycon. You have to rearrange the controllers in the config so that the full controller is player one, then it should work
